### PR TITLE
QDK Dec 2021 release

### PR DESCRIPTION
**Pull request by vxfield**

### CODE REVIEW

File: samples/algorithms/variational-algorithms/environment.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ channels:
 dependencies:
   - notebook
   - matplotlib
-  - qsharp==0.21.2111.177148
+  - qsharp==0.21.2112180703
```
Comment (cgranade) [inline samples/algorithms/variational-algorithms/environment.yml:11]: It looks like the extra dot got dropped here? Please double-check all four environment.yml files to make sure that they work with the new package versions. Thank you!

**Review by cgranade — APPROVED** [review]
Approved with one comment that should be addressed before merging. Thanks!

File: samples/algorithms/variational-algorithms/environment.yml (hunk 1 above)
Comment (vxfield) [reply samples/algorithms/variational-algorithms/environment.yml]: Thanks, Chris.
I have also fixed the Bash script to correctly format the Nuget, Python and Docker versions regardless of how the input version is given.

Comment (vxfield) [reply samples/algorithms/variational-algorithms/environment.yml]: I'll also update the QDK repo (which may be overriding this script) with the correct formatting of versions.

**Pull request merged**